src/gui/modals/FileDialog.cpp

```diff
@@ -45,19 +47,40 @@ FileDialog::FileDialog( QWidget *parent, const QString &caption,
 
 	setOption( QFileDialog::DontUseNativeDialog );
 
-	// Add additional locations to the sidebar
+#ifdef LMMS_BUILD_LINUX
+	QList<QUrl> urls;
+#endif
+
+#ifndef LMMS_BUILD_LINUX
 	QList<QUrl> urls = sidebarUrls();
-	urls << QUrl::fromLocalFile( QStandardPaths::writableLocation( QStandardPaths::DesktopLocation ) );
-	// Find downloads directory
-	QDir downloadDir( QDir::homePath() + "/Downloads" );
-	if ( ! downloadDir.exists() )
-		downloadDir.setPath(QStandardPaths::writableLocation( QStandardPaths::DownloadLocation ));
-	if ( downloadDir.exists() )
-		urls << QUrl::fromLocalFile( downloadDir.absolutePath() );
+#endif
+
+	QDir desktopDir;
+	desktopDir.setPath(QStandardPaths::writableLocation(QStandardPaths::DesktopLocation));
+	if (desktopDir.exists())
```

Comment:
Why not construct `desktopDir` directly with the path? Also, you can scope the variable to the if-statement in C++17:
```suggestion
	if (const auto desktopDir = QDir{QStandardPaths::writableLocation(QStandardPaths::DesktopLocation)};
		desktopDir.exists())
```
Similarly for the other directories here.

Reply:
I am not aware of this coding feature and find it less readable. But if this is best practice I could change.

Reply:
Feel free to leave it if you'd rather not use it. I personally like it as it reduces the scope of the variable, which I find easier to read as I know where the variable is relevant with more precision. But the code needs to be readable by everyone, so if you're more comfortable not doing this, leave it. The scope is not particularly important in this case - it's not a mutex lock or anything like that, which is where this feature really comes in handy. I do prefer passing the path to `QDir`'s constructor though, if you're happy just to do that bit.